src/frontend/web_application/src/modules/draftMessage/components/IdentitySelector/presenter.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Trans } from '@lingui/macro'; // eslint-disable-line import/no-extraneous-dependencies
+import { AdvancedSelectFieldGroup, Icon } from '../../../../components';
+import { PROVIDER_EMAIL, PROVIDER_GMAIL, PROVIDER_TWITTER, ProviderIcon } from '../../../remoteIdentity';
```

Comment:
if we only use dynamic list of protocols, it won't work because all must be dynamic (details, explanations, i18n, icons for all providers) and it is almost sure somewhere we will write hardcoded name for a specific behavior (like in this component when selecting the provider icon) which would be very bad.
So I prefer going slowly using constants that would easier to find and refactor when we would be sure about this.

Reply:
(btw in theory a UI for a ReST API could be fully dynamic ... but it ain't gonna happen, it's basically a scientist delusion ;) )

Reply:
this is what I meant ;)

```
isMail = () => {
    const { message } = this.props;

    return message.type === 'email';
  };
```